lib/matplotlib/axes/_base.py

```diff
@@ -1271,11 +1271,17 @@ def set_aspect(self, aspect, adjustable=None, anchor=None, share=False):
         if not (cbook._str_equal(aspect, 'equal')
                 or cbook._str_equal(aspect, 'auto')):
             aspect = float(aspect)  # raise ValueError if necessary
+
+        if cbook._str_equal(aspect, 'equal') and self.name == '3d':
+            raise NotImplementedError(
+                'It is not currently possible to have equal aspect on 3D axes')
```

Comment:
Equal aspect is the same as aspect = 1 (according to the docs).  So I guess this mean that any numeric aspect is likely wrong, and should likely trigger an exception, as well?

Reply:
Changed it to just check if anything that isn't 'auto' is given.